### DISCUSSION TITLE
404 NotFound server(SSR) handler

### DISCUSSION THREAD

**Discussion by mhf-ir** [nuxt/nuxt#16247] [Questions]
Any guide or document for 404 handler on server. on SSR.

```js

// sample of fastify
// https://www.fastify.io/docs/latest/Reference/Server/#setnotfoundhandler

app.setNotFoundHandler(function async (request, reply) {
    const newURL = await httpClient('https://server.tld/api/found-redirect', 'POST', { url: request.url });
    if (newURL) {
      // do redirect
    } else {
      // insert into log system for check later
      // and throw typical 404 response
    }
  });

```

Use case many of 404 URL seen by end user via google search console or another mechanism and with interface user can add it into database for process.

In case of URL of page change by any regex or even slug. better to handle in this layer.

**Comment by TheAlexLichter** [comment]
Hey!

You can use a middleware together with e.g. [`createError`](https://nuxt.com/docs/api/utils/create-error#throwing-errors-in-api-routes) to throw an error ☺️

Also, you can use an [`error.vue`](https://nuxt.com/docs/getting-started/error-handling#rendering-an-error-page) page to catch errors
